### PR TITLE
Add Managed Headers resource

### LINKED ISSUE #1607: Feature: Support for configuring `managed_headers`
### Current Terraform and Cloudflare provider version

2.27

### Description

Managed Transforms do not appear to be a currently supported or documented item in the Terraform provider, but are generally available via the Cloudflare UI and API (https://developers.cloudflare.com/rules/transform/managed-transforms/configure/ / https://api.cloudflare.com/#managed-headers-api-change-state-of-managed-headers)

### Use cases

The ability to enabling / disable specific of Managed Transforms via Terraform would be excellent.

### Potential Terraform configuration

```hcl
resource "cloudflare_ruleset" "zone_level_managed_waf" {
  zone_id     = "cb029e245cfdd66dc8d2e570d5dd3322"
  name        = "managed transforms"
  description = "managed transforms ruleset description"
  kind        = "zone"
  phase       = "http_request_transform_managed"

  rules {
    action = "execute"
    action_parameters {
      id = "add_cf-bot-score_header"
    }
    description = "Execute Cloudflare Managed Ruleset on my zone-level phase entry point ruleset"
    enabled = true
  }
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by WowVeryLogin**
Adding support of the Managed Headers API. Tests and example resource are present.

Closes #1607

### CODE REVIEW

File: internal/provider/resource_cloudflare_managed_headers.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareManagedHeadersImport,
+		},
+		SchemaVersion: 0,
+		Description: `
+The [Cloudflare Managed Headers](https://developers.cloudflare.com/rules/transform/managed-transforms/)
+allows you to add or remove some predefined headers to one's requests or origin responses.
+
+~> **NOTE:** You can configure Managed Headers using the dashboard (https://api.cloudflare.com/#managed-headers-api-properties)
+Terraform will override yours configuration if it exists.`,
```
Comment (jacobbednarz) [inline internal/provider/resource_cloudflare_managed_headers.go:29]:
```suggestion
Terraform will override your configuration if it exists.`,
```

**Comment by jacobbednarz** [comment]
❤️ ❤️ ❤️ this! couple of CI lint failures, removing the dep bumps from this PR, and once you run `make docs`, this is ready to land. thanks for such a quick turn around.

File: internal/provider/resource_cloudflare_managed_headers.go (hunk 1 above)
Comment (WowVeryLogin) [reply internal/provider/resource_cloudflare_managed_headers.go]: Fixed

File: docs/resources/cloudflare_ruleset.md [hunk 2]
```diff
@@ -286,7 +286,7 @@ resource "cloudflare_ruleset" "custom_fields_logging_example" {
...
 - `name` (String) Name of the ruleset.
-- `phase` (String) Point in the request/response lifecycle where the ruleset will be created. Available values: `"ddos_l4"`, `"ddos_l7"`, `"http_log_custom_fields"`, `"http_request_firewall_custom"`, `"http_request_firewall_managed"`, `"http_request_late_transform"`, `"http_request_main"`, `"http_request_sanitize"`, `"http_request_transform"`, `"http_request_origin"`, `"http_response_firewall_managed"`, `"http_response_headers_transform"`, `"magic_transit"`, `"http_ratelimit"`, `"http_request_sbfm"`.
+- `phase` (String) Point in the request/response lifecycle where the ruleset will be created. Available values: `"ddos_l4"`, `"ddos_l7"`, `"http_log_custom_fields"`, `"http_request_firewall_custom"`, `"http_request_firewall_managed"`, `"http_request_late_transform"`, `"http_request_late_transform_managed"`, `"http_request_main"`, `"http_request_origin"`, `"http_request_sanitize"`, `"http_request_transform"`, `"http_response_firewall_managed"`, `"http_response_headers_transform"`, `"magic_transit"`, `"http_ratelimit"`, `"http_request_sbfm"`.
```
Comment (jacobbednarz) [inline docs/resources/cloudflare_ruleset.md:289]: oh nice! a bump in `cloudflare-go` automatically gave us updated documentation 🎉

File: internal/provider/resource_cloudflare_managed_headers_test.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	resourceName := "cloudflare_managed_headers." + rnd
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareManagedHeaders(rnd, zoneID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(resourceName, "managed_request_headers.#", "1"),
```
Comment (jacobbednarz) [inline internal/provider/resource_cloudflare_managed_headers_test.go:82]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "managed_request_headers.#", "4"),
```

Comment (WowVeryLogin) [reply internal/provider/resource_cloudflare_managed_headers_test.go]: Fixed

File: internal/provider/resource_cloudflare_managed_headers_test.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		ZoneID: zoneID,
+	})
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("Failed to disable Cloudflare Zone Managed Headers: %s", err))
+	}
+
+	return nil
+}
+
+func TestAccCloudflareManagedHeaders(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (WowVeryLogin) [reply internal/provider/resource_cloudflare_managed_headers_test.go:68]: Yes, fixed the test

**Comment by jacobbednarz** [comment]
CI is now happy with a shimed in cloudflare-go 

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareManagedHeaders" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareManagedHeaders
=== PAUSE TestAccCloudflareManagedHeaders
=== CONT  TestAccCloudflareManagedHeaders
--- PASS: TestAccCloudflareManagedHeaders (13.04s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/provider	13.478s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

**Pull request merged**